streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -67,58 +69,151 @@
 
     private static final String TOSTREAM_NAME = "KTABLE-TOSTREAM-";
 
+    public static final String STATE_STORE_NAME = "STATE-STORE-";
+
     private final ProcessorSupplier<?, ?> processorSupplier;
 
-    private final String storeName;
+    private final String queryableName;
 
     private boolean sendOldValues = false;
+    private final Serde<K> keySerde;
+    private final Serde<V> valSerde;
 
+    public KTableImpl(KStreamBuilder topology,
+                      String name,
+                      ProcessorSupplier<?, ?> processorSupplier,
+                      Set<String> sourceNodes,
+                      final String queryableName) {
+        super(topology, name, sourceNodes);
+        this.processorSupplier = processorSupplier;
+        this.queryableName = queryableName;
+        this.keySerde = null;
+        this.valSerde = null;
+    }
 
     public KTableImpl(KStreamBuilder topology,
                       String name,
                       ProcessorSupplier<?, ?> processorSupplier,
+                      final Serde<K> keySerde,
+                      final Serde<V> valSerde,
                       Set<String> sourceNodes,
-                      final String storeName) {
+                      final String queryableName) {
         super(topology, name, sourceNodes);
         this.processorSupplier = processorSupplier;
-        this.storeName = storeName;
+        this.queryableName = queryableName;
+        this.keySerde = keySerde;
+        this.valSerde = valSerde;
     }
 
     @Override
-    public String getStoreName() {
-        return this.storeName;
+    public String getQueryableName() {
+        return this.queryableName;
     }
 
+
     @Override
-    public KTable<K, V> filter(Predicate<? super K, ? super V> predicate) {
+    public KTable<K, V> filter(final Predicate<? super K, ? super V> predicate) {
```

Comment:
Can we make this function delegate to the one that takes a store name?
i.e, in the other one we just do
```
if(queryableName != null) {
   final StateStoreSupplier ...
   topology.addStateStore(...)
}
```